pkg/operator/controller/certificate/default_cert.go

```diff
@@ -17,67 +17,73 @@ import (
 
 // ensureDefaultCertificateForIngress creates or deletes an operator-generated
 // default certificate for a given IngressController as appropriate.  Returns true
-// if it creates, updates, or deletes the secret for the certificate, false
+// if it creates or deletes the secret for the certificate, false
 // otherwise.
 func (r *reconciler) ensureDefaultCertificateForIngress(caSecret *corev1.Secret, namespace string, deploymentRef metav1.OwnerReference, ci *operatorv1.IngressController) (bool, error) {
 	ca, err := crypto.GetCAFromBytes(caSecret.Data["tls.crt"], caSecret.Data["tls.key"])
 	if err != nil {
 		return false, fmt.Errorf("failed to get CA from secret %s/%s: %v", caSecret.Namespace, caSecret.Name, err)
 	}
-	desired, err := desiredRouterDefaultCertificateSecret(ca, namespace, deploymentRef, ci)
+	wantCert, desired, err := desiredRouterDefaultCertificateSecret(ca, namespace, deploymentRef, ci)
 	if err != nil {
 		return false, err
+	} else if !wantCert {
+		// If the opeartor generated certificate is not being used, ensure that the ingress controller's
+		// Spec.DefaultCertificate secret exists before deleting the operator generated secret.
+		err := r.lookupUserSpecifiedRouterDefaultCertificate(ci, namespace)
+		if err != nil {
+			return false, fmt.Errorf("failed to lookup user specified default certificate: %v", err)
+		}
 	}
-	current, err := r.currentRouterDefaultCertificate(ci, namespace)
+
+	haveCert, current, err := r.currentRouterDefaultCertificate(ci, namespace)
 	if err != nil {
 		return false, err
 	}
 	switch {
-	case desired == nil && current == nil:
+	case !wantCert && !haveCert:
 		// Nothing to do.
-	case desired == nil && current != nil:
+	case !wantCert && haveCert:
 		if deleted, err := r.deleteRouterDefaultCertificate(current); err != nil {
 			return false, fmt.Errorf("failed to delete default certificate: %v", err)
 		} else if deleted {
 			r.recorder.Eventf(ci, "Normal", "DeletedDefaultCertificate", "Deleted default wildcard certificate %q", current.Name)
 			return true, nil
 		}
-	case desired != nil && current == nil:
+	case wantCert && !haveCert:
 		if created, err := r.createRouterDefaultCertificate(desired); err != nil {
 			return false, fmt.Errorf("failed to create default certificate: %v", err)
 		} else if created {
 			r.recorder.Eventf(ci, "Normal", "CreatedDefaultCertificate", "Created default wildcard certificate %q", desired.Name)
 			return true, nil
 		}
-	case desired != nil && current != nil:
-		// TODO Update if CA certificate changed.
```

Comment:
From the PR description:

> Finally, this commit removes a stale TODO comment requesting that update logic be added to
> default_cert.go. Comparing the results of `desiredRouterDefaultCertificateSecret` to
> `currentRouterDefaultCertificate` would be pointless since `desiredRouterDefaultCertificateSecret`
> always returns a uniquely generated certificate.

It is true that comparing the results of `desiredRouterDefaultCertificateSecret` to `currentRouterDefaultCertificate` would be pointless, but we could still check if the CA certificate matches or is newer than the serving certificate. .

Anyway, I think we should return `true, nil` here.

Reply:
Ah, right. 
For now I'll leave the TODO and return true, nil.